app/elm/Images/GalleryView.elm

```diff
@@ -0,0 +1,74 @@
+module Images.GalleryView exposing (galleryView)
+
+import Html exposing (Html, div, a, img, text)
+import Html.Attributes exposing (class, href, src, alt, title)
+import Routing exposing (imagePath, personPath, personImagePath)
+import Images.Models exposing (..)
+import Images.Utils exposing (..)
+
+
+galleryView : Maybe PersonId -> ( List Image, List Person ) -> Html msg
+galleryView maybePersonId ( allImages, allPeople ) =
+    case maybePersonId of
+        Just personId ->
+            case (getPerson personId allPeople) of
+                Just person ->
+                    (getImagesOfPerson personId allImages)
+                        |> List.map (cardView maybePersonId)
+                        |> buildGalleryView
+
+                Nothing ->
+                    notFoundView
+
+        Nothing ->
+            allImages
+                |> List.map (cardView Maybe.Nothing)
+                |> buildGalleryView
+
+
+buildGalleryView : List (Html msg) -> Html msg
+buildGalleryView cards =
+    div
+        [ class "gallery" ]
+        cards
+
+
+cardView : Maybe PersonId -> Image -> Html msg
+cardView maybePersonId image =
+    case maybePersonId of
+        Just personId ->
+            buildCardView
+                (personImagePath image.id personId)
+                image
+
+        Nothing ->
+            buildCardView
+                (imagePath image.id)
+                image
+
+
+buildCardView : String -> Image -> Html msg
+buildCardView path image =
+    a
+        [ class "view-image"
+        , href path
+        ]
+        [ renderThumbnail image ]
```

Comment:
I think you could tighten up `cardView` and `buildCardView` into a single function by doing something like:

```elm
cardView : Maybe PersonId -> Image -> Html msg
cardView maybePersonId image =
    let
        path =
            maybePersonId
                |> Maybe.map (personImagePath image.id)
                |> Maybe.withDefault (imagePath image.id)
    in
        a
            [ class "view-image"
            , href path
            ]
            [ renderThumbnail image ]
```
Although I could see where `buildCardView` might also be used generically in future contexts, which might be a reason to keep them separate.